pkg/master/controller.go

```diff
@@ -79,6 +79,15 @@ type Controller struct {
 
 // NewBootstrapController returns a controller for watching the core capabilities of the master
 func (c *completedConfig) NewBootstrapController(legacyRESTStorage corerest.LegacyRESTStorage, serviceClient coreclient.ServicesGetter, nsClient coreclient.NamespacesGetter, eventClient coreclient.EventsGetter) *Controller {
+	publicServicePort := 443 // this is the legacy behaviour. Without a secure port, we cannot serve the kubernetes service correctly though.
```

Comment:
This seems weird.  If we require it, why not simply fail early if we have no secure port?

Reply:
just replicating the old behaviour here. Am fine with failing as well. In other words: do not allow zero with `--secure-port`.

Reply:
What's the behavior previously when `--secure-port` is set to 0?

Reply:
Then https://github.com/kubernetes/kubernetes/pull/65830/files#diff-01f8ed603e3f5cc358d0394c0ab0cb00L51 was skipped and the default value of 443 was staying.

Reply:
Fixed